tests/connectionpool_test.go

```diff
@@ -137,3 +143,193 @@ func TestConnectionPool(t *testing.T) {
+					CloudName: cfg.PrimaryCloudName,
+				},
+			},
+		},
+	}
+
+	dbObj := &v1alpha1.Database{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dbName,
+			Namespace: defaultNamespace,
+		},
+		Spec: v1alpha1.DatabaseSpec{
+			ServiceDependant: v1alpha1.ServiceDependant{
+				ProjectDependant: v1alpha1.ProjectDependant{
+					ProjectField: v1alpha1.ProjectField{
+						Project: cfg.Project,
+					},
+					AuthSecretRefField: v1alpha1.AuthSecretRefField{
+						AuthSecretRef: &v1alpha1.AuthSecretReference{
+							Name: secretRefName,
+							Key:  secretRefKey,
+						},
+					},
+				},
+				ServiceField: v1alpha1.ServiceField{
+					ServiceName: pgName,
+				},
+			},
+		},
+	}
+
+	poolObj := &v1alpha1.ConnectionPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      poolName,
+			Namespace: defaultNamespace,
+		},
+		Spec: v1alpha1.ConnectionPoolSpec{
+			ServiceDependant: v1alpha1.ServiceDependant{
+				ProjectDependant: v1alpha1.ProjectDependant{
+					ProjectField: v1alpha1.ProjectField{
+						Project: cfg.Project,
+					},
+					AuthSecretRefField: v1alpha1.AuthSecretRefField{
+						AuthSecretRef: &v1alpha1.AuthSecretReference{
+							Name: secretRefName,
+							Key:  secretRefKey,
+						},
+					},
+				},
+				ServiceField: v1alpha1.ServiceField{
+					ServiceName: pgName,
+				},
+			},
+			DatabaseName: dbName,
+			// No username field
+			PoolSize: 10,
+			PoolMode: "transaction",
+		},
+	}
+	require.NoError(t, s.ApplyObjects(pgObj, dbObj, poolObj))
+	require.NoError(t, s.GetRunning(pgObj, pgName))
+	require.NoError(t, s.GetRunning(dbObj, dbName))
+	require.NoError(t, s.GetRunning(poolObj, poolName))
+
+	avnPool := findPoolFunc(cfg.Project, pgName, poolName)
+	require.NotNil(t, avnPool, "connection pool should be created in Aiven")
+
+	assert.Equal(t, pgName, poolObj.Spec.ServiceName)
+	assert.Equal(t, poolName, poolObj.GetName())
+	assert.Equal(t, poolName, avnPool.PoolName)
+	assert.Equal(t, dbName, poolObj.Spec.DatabaseName)
+	assert.Equal(t, dbName, avnPool.Database)
+	assert.Empty(t, poolObj.Spec.Username) // username should be empty in the spec
+
+	secret, err := s.GetSecret(poolObj.GetName())
+	require.NoError(t, err)
+
+	// Check secret has necessary connection info
+	assert.NotEmpty(t, secret.Data["CONNECTIONPOOL_HOST"])
+	assert.NotEmpty(t, secret.Data["CONNECTIONPOOL_PORT"])
+	assert.NotEmpty(t, secret.Data["CONNECTIONPOOL_DATABASE"])
+	assert.NotEmpty(t, secret.Data["CONNECTIONPOOL_USER"]) // For inbound user, this is the default user
+	assert.NotEmpty(t, secret.Data["CONNECTIONPOOL_PASSWORD"])
+	assert.NotEmpty(t, secret.Data["CONNECTIONPOOL_SSLMODE"])
+	assert.NotEmpty(t, secret.Data["CONNECTIONPOOL_DATABASE_URI"])
+	assert.NotEmpty(t, secret.Data["CONNECTIONPOOL_CA_CERT"])
+
+	// Update the connection pool
+	poolObj.Spec.PoolSize = 25
+	poolObj.Spec.PoolMode = "session"
+	require.NoError(t, err)
+	require.NoError(t, s.ApplyObjects(poolObj))
+	require.NoError(t, s.GetRunning(poolObj, poolName))
```

Comment:
Thanks for adding update test. Considering [the issue](https://github.com/aiven/aiven-operator/pull/905#discussion_r2001201380) from the previous commit, I highly recommend to add an additional check that there is only one pool created.